### PR TITLE
feat: implement getMessageStatus

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Implements the `getMessageStatus` function for the SDK.

**Metadata**
- Fixes ENG-1799

### CODE REVIEW

File: packages/sdk/src/cross-chain-provider.ts [hunk 1]
```diff
@@ -436,4 +479,169 @@ export class CrossChainProvider implements ICrossChainProvider {
...
+    ).toNumber()
+  }
+
+  public async getChallengePeriodBlocks(): Promise<number> {
+    return Math.ceil(
+      (await this.getChallengePeriodSeconds()) / this.l1BlockTime
+    )
+  }
+
+  public async getMessageStateRoot(
```
Comment (smartcontracts) [inline packages/sdk/src/cross-chain-provider.ts:495]: Everything from here down is basically copy/pasted from here: https://github.com/ethereum-optimism/optimism/blob/develop/packages/message-relayer/src/relay-tx.ts. I'll add tests for this at a later point in time.

**Comment by tynes** [comment]
Does this need a changeset?

File: packages/sdk/src/cross-chain-provider.ts [hunk 2]
```diff
@@ -436,4 +479,169 @@ export class CrossChainProvider implements ICrossChainProvider {
   ): Promise<number> {
     throw new Error('Not implemented')
   }
+
+  public async getChallengePeriodSeconds(): Promise<number> {
+    return BigNumber.from(
```
Comment (tynes) [inline packages/sdk/src/cross-chain-provider.ts:484]: Why does this need `BigNumber.from` and then `toNumber()`?

File: packages/sdk/test/cross-chain-provider.spec.ts [hunk 3]
```diff
@@ -783,46 +760,204 @@ describe('CrossChainProvider', () => {
   })
 
   describe('getMessageStatus', () => {
+    let scc: Contract
+    let l1Messenger: Contract
+    let l2Messenger: Contract
+    let provider: CrossChainProvider
+    beforeEach(async () => {
+      scc = (await (await ethers.getContractFactory('MockSCC')).deploy()) as any
```
Comment (tynes) [inline packages/sdk/test/cross-chain-provider.spec.ts:768]: nit: nested `await`s defeat the purpose of having `await` - making async code very readable

File: packages/sdk/test/cross-chain-provider.spec.ts [hunk 4]
```diff
@@ -783,46 +760,204 @@ describe('CrossChainProvider', () => {
...
+              MessageStatus.READY_FOR_RELAY
+            )
+          })
         })
       })
     })
 
     describe('when the message does not exist', () => {
-      it('should throw an error')
+      it('should throw an error', async () => {
```
Comment (tynes) [inline packages/sdk/test/cross-chain-provider.spec.ts:958]: I feel like it should not throw an error but instead return `null`. Can you `skip` this test to make sure we know its not implemented when running the tests?

**Comment by smartcontracts** [comment]
> Does this need a changeset?

We are not publishing this package publicly yet so I have not been adding changesets (to avoid the npm publish).

File: packages/sdk/test/cross-chain-provider.spec.ts (hunk 3 above)
Comment (smartcontracts) [reply packages/sdk/test/cross-chain-provider.spec.ts:768]: I did this in other places, I'll make a comment with a TODO and turn this into a good first issue for someone to get rid of these nested awaits everywhere within this package.

File: packages/sdk/src/cross-chain-provider.ts [hunk 5]
```diff
@@ -48,18 +51,24 @@ export class CrossChainProvider implements ICrossChainProvider {
...
     l2Provider: ProviderLike
     l1ChainId: NumberLike
+    l1BlockTime?: NumberLike
     contracts?: DeepPartial<OEContractsLike>
     bridges?: Partial<CustomBridgesLike>
   }) {
     this.l1Provider = toProvider(opts.l1Provider)
     this.l2Provider = toProvider(opts.l2Provider)
     this.l1ChainId = toBigNumber(opts.l1ChainId).toNumber()
+    this.l1BlockTime = opts.l1BlockTime
```
Comment (smartcontracts) [inline packages/sdk/src/cross-chain-provider.ts:69]: Have default be based on chain ID

**Review by SmitV — APPROVED** [review]

**Pull request merged**